Content.Shared/Doors/Systems/PryingSystem.cs

```diff
@@ -0,0 +1,131 @@
+using Content.Shared.Doors.Prying.Components;
+using Content.Shared.DoAfter;
+using Robust.Shared.Serialization;
+using Content.Shared.Administration.Logs;
+using Content.Shared.Database;
+using Content.Shared.Doors.Components;
+using Content.Shared.Doors.Systems;
+using Content.Shared.Tools.Components;
+
+namespace Content.Shared.Doors.Prying.Systems;
+public class DoorPryingSystem : EntitySystem
+{
+    [Dependency] private readonly ISharedAdminLogManager _adminLog = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+    [Dependency] private readonly SharedDoorSystem _doorSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<DoorComponent, DoorPryDoAfterEvent>(OnDoAfter);
+    }
+
+    /// <summary>
+    ///     Pry open a door.
+    /// </summary>
+    public bool TryPry(EntityUid target, EntityUid user, DoorComponent door, out DoAfterId? id, EntityUid tool)
+    {
+        id = null;
+
+        DoorPryingComponent? comp = null;
+        if (!Resolve(tool, ref comp))
+            return false;
+
+        if (comp.NeedsComponent)
+        {
+            ToolComponent? tComp = null;
+            if (!Resolve(tool, ref tComp) || !tComp.Qualities.Contains("Prying"))
+                return false;
+        }
+
+        if (!CanPry(target, user, comp.PryPowered))
+        {
+            return true;
+        }
+
+        StartPry(target, user, door, tool, comp.SpeedModifier, out id);
+        return true;
+    }
+
+    /// <summary>
+    ///     Pry open a door.
+    /// </summary>
+    public bool TryPry(EntityUid target, EntityUid user, DoorComponent door, out DoAfterId? id)
```

Comment:
Is this id supposed to be marked as notnullwhen(true)

Reply:
No, I was just missing some documentation that stated that it returned true to mark an interactusingevent as handled even if a doafter had not occurred in some cases.